hp/elitebook/830/g6/default.nix

```diff
@@ -0,0 +1,26 @@
+{ pkgs, lib, ... }:
+{
+  imports = [
+    ../../../../common/cpu/intel
+    ../../../../common/pc/laptop
+    ../../../../common/pc/laptop/acpi_call.nix
+    ../../../../common/pc/laptop/ssd
+  ];
+
+  # Needed for wifi
+  hardware.enableRedistributableFirmware = lib.mkDefault true;
+
+  # Cooling management
+  services.thermald.enable = lib.mkDefault true;
+
+  # Allows for updating firmware via `fwupdmgr`.
+  services.fwupd.enable = lib.mkDefault true;
+
+  # Enables ACPI platform profiles
+  boot = lib.mkIf (lib.versionAtLeast pkgs.linux.version "6.1") {
+    kernelModules = [ "hp-wmi" ];
+  };
+
+  # reduces warnings/errors in boot log, seems to eliminate the ocassional boot hangs described in readme
+  hardware.enableAllFirmware = lib.mkDefault true;
```

Comment:
I dropped `nixpkgs.config.allowUnfree` again as it's not compatible with our testing infrastructure, where we pass in a shared nixpkgs instance for performance reason. The same trick is also used by larger NixOS installations. Furthermore it might be a bit sneaky to enable allowUnfree i.e. not something a user would expect a module to do.
The error thrown by `hardware.enableAllFirmware` on the other hand however should already good enough for users to know that they also have to enable `allowUnfree` in their code.